File: frontend/src/types/prakriti.ts
export interface PrakritiOption {
  id: string;
  text: string;
  vata: number;
  pitta: number;
  kapha: number;
}

export interface PrakritiQuestion {
  id: string;
  category: 'Physical' | 'Physiological' | 'Psychological';
  title: string;
  options: PrakritiOption[];
}

export interface RadarDataPoint {
  subject: string;
  A: number;
  fullMark: number;
}

export interface PrakritiResult {
  id?: number;
  encounter_id?: number;
  vata_score: number;
  pitta_score: number;
  kapha_score: number;
  vata_percent: number;
  pitta_percent: number;
  kapha_percent: number;
  primary_prakriti: string;
  secondary_prakriti?: string;
  radar_data?: RadarDataPoint[];
  responses_json?: string;
  notes?: string;
}

export interface VikritiResult {
  id?: number;
  encounter_id?: number;
  vata_score: number;
  pitta_score: number;
  kapha_score: number;
  vata_percent: number;
  pitta_percent: number;
  kapha_percent: number;
  dominant_pattern: string;
  dosha_distribution?: Array<{ name: string; percentage: number; fill: string }>;
  notes?: string;
}

File: backend/app/schemas/prakriti.py
from typing import Optional, Dict, Any, List
from pydantic import BaseModel

class PrakritiCalculateRequest(BaseModel):
    encounter_id: Optional[int] = None
    responses: Dict[str, str]  # question_id -> option_id
    notes: Optional[str] = None

class PrakritiResponse(BaseModel):
    id: Optional[int] = None
    encounter_id: Optional[int] = None
    vata_score: float
    pitta_score: float
    kapha_score: float
    vata_percent: float
    pitta_percent: float
    kapha_percent: float
    primary_prakriti: str
    secondary_prakriti: Optional[str] = None
    radar_data: Optional[List[Dict[str, Any]]] = None
    responses_json: Optional[str] = None
    notes: Optional[str] = None

    class Config:
        from_attributes = True

class VikritiCalculateRequest(BaseModel):
    encounter_id: Optional[int] = None
    symptoms: Optional[List[Dict[str, Any]]] = []
    ashtavidha: Optional[Dict[str, Any]] = {}
    notes: Optional[str] = None

class VikritiResponse(BaseModel):
    id: Optional[int] = None
    encounter_id: Optional[int] = None
    vata_score: float
    pitta_score: float
    kapha_score: float
    vata_percent: float
    pitta_percent: float
    kapha_percent: float
    dominant_pattern: str
    dosha_distribution: Optional[List[Dict[str, Any]]] = None
    notes: Optional[str] = None

    class Config:
        from_attributes = True
